packages/idos-sdk-js/src/__tests__/kwil-wrapper.test.ts

```diff
@@ -0,0 +1,79 @@
+import { Utils, WebKwil } from "@kwilteam/kwil-js";
+import { Wallet } from "ethers";
+import { implicitAddressFromPublicKey } from "src/lib";
+import { KwilWrapper } from "src/lib/kwil-wrapper";
+import { beforeEach, describe, expect, test } from "vitest";
+
+const kwilProvider = "kwil-provider";
+const chainId = "chain-id";
+const dbId = "db-id";
+
+class TestKwilClient extends WebKwil {
+  constructor() {
+    super({ kwilProvider, chainId });
+  }
+}
+
+let kwilWrapper: KwilWrapper;
+
+describe("kwil-wrapper", () => {
+  beforeEach(() => {
+    kwilWrapper = new KwilWrapper(new TestKwilClient(), kwilProvider, dbId);
+  });
+
+  test("should create a new instance of KwilWrapper", () => {
+    expect(kwilWrapper.client).toBeInstanceOf(WebKwil);
+    expect(kwilWrapper.kwilProvider).toBe(kwilProvider);
+    expect(kwilWrapper.dbId).toBe(dbId);
+  });
+
+  test("should set a `secp256k1_ep` signer", async () => {
+    const signer: any = Wallet.createRandom();
+
+    await kwilWrapper.setSigner({
+      accountId: signer.address,
+      signer,
+      signatureType: "secp256k1_ep",
+    });
+
+    expect(kwilWrapper.signer?.signatureType).toBe("secp256k1_ep");
+  });
+
+  test("should set a `nep413` signer", async () => {
+    const signer = () => Promise.resolve(new Uint8Array());
+    const pk = "ed25519:ACWH3UoS3JVnVhmvHMghKN9K76HqiYQpwodJbJCUtciY";
+    const accountId = implicitAddressFromPublicKey(pk);
+    await kwilWrapper.setSigner({
+      accountId,
+      signer,
+      signatureType: "nep413",
+    });
+
+    expect(kwilWrapper.signer?.signatureType).toBe("nep413");
+  });
```

Comment:
Meh. Not opposed, but it feels like too much testing the wires, and not enough testing the behavior. Unsure what that would even look like.